### DISCUSSION TITLE
Introductions

### DISCUSSION THREAD

**Discussion by yrashk** [pgedc/pgedc#20] [General]
Greetings, fellow Postgres extension developers! 

First of all, thank you for joining the coalition. It's great that so many people support and subscribe to the mission.

Let's get to know each other! Please use this discussion to introduce yourself, your work and whatever else you deem important.

---

cc @pgedc/members

**Comment by yrashk** [comment]
I'll start with myself to set this in motion.

👋 My name is Yurii. I have been a software developer since a young age and an entrepreneur for most of my adult life. I've always been interested in the fabric that makes software systems. There are no programs without data, so databases of different kinds were always in my purview, and I always tried to find the next most interesting thing about them. This journey has led me to start [Omnigres](https://github.com/omnigres/omnigres), where I've developed many extensions. I've been recently speaking about extensions and the experiences of building them at Postgres events. Outside of the Postgres world, I program in various languages and am always keen to learn more about computer science. 

One thing that excites me about extensions is that they bring the model of database systems a notch closer to the conceptual model of sophisticated computational engines, where durability is one of its capabilities. This can profoundly affect the quality of software and businesses that use it.

I am also an amateur marathon runner, hiker and cyclist!

**Comment by abigalekim** [comment]
Hi! My name is Abi! I am an all-things databases enthusiast. In the past, I have worked on Redshift, YugabyteDB, and NoisePage. Currently, I work at TileDB on their array DBMS and research DBMS extensibility. My research on DBMS extensibility consists of two major parts. First, it has resulted in a taxonomy of existing DBMS extensibility design. Second, I developed a PostgreSQL extensions analyzer that tests the incompatibility of different extensions with the existing system and each other. My studies have led me to believe that PostgreSQL extensibility has a lot of potential to have a lasting impact on the DBMS extensibility sphere, which is what brings me here.

I'm also passionate about things on the research side compared to industry, although I really enjoy seeing research have real-world practical impact. As a result, I've completed a Master's degree with thesis component at CMU, and will start a PhD in database systems at UW Madison in the fall!

Hobbies of mine include fiber crafts (knitting, crochet) and many different ways of exercise (pickleball and hiking are faves right now).

**Comment by mkindahl** [comment]
Hi fellow open-sourcers!

I have been working with computers for a long time and only "recently" started working with database systems when I joined MySQL in 2004. I spent many years working with MySQL and went through two acquisitions: first joining Sun Microsystems and then Oracle when Sun was bought by Oracle. In my time at MySQL/Sun/Oracle I worked mainly on replication and high-availability, but also started and lead a few teams. Before that I worked as a researcher of formal verification methods and as a compiler implementer.

My experience with PostgreSQL started when I joined Timescale in 2019 and started working with my current team. Right now I am working as a database architect with a particular interest in the table and index access methods, which I see an a critical enabler for even wider adoption of PostgreSQL.

I am an open-source enthusiast since Eric Raymond published _The Cathedral and the Bazaar_ and my first open-source contribution was to a formal verification tool called Caesar in my first years as a research student.

I usually spend my personal time reading and learning about mathematics (in particular statistics and combinatorics), computer science, and economics. My favorite outdoor activities are jogging, swimming, or kayaking.

**Comment by tsg** [comment]
Hey, I am Tudor. I am a software developer interested in databases, distributed systems, and systems level programming in general. I am the CTO of [xata.io](https://xata.io/), we are a Postgres platform bringing in some goodies on top of Postgres: replication to Elasticsearch, dev branches, zero-downtime schema migrations, and more. 

We have recently open-sourced [pgzx](https://github.com/xataio/pgzx), which is a framework for building Postgres extensions with Zig. You can think of it as similar to pgrx, but for Zig, and in the next couple of weeks I'm planning to add more example extensions to it. I'd love to see others build on top of it, Zig is an unconventional language choice at this point, but when it comes to Postgres extensions it can be a middle ground between C and Rust.

I like reading and all sorts of sports. I've recently learned to solve the Rubik cube and I'm getting into speedcubing for fun.

**Comment by marcoslot** [comment]
Hi! I'm Marco, I'm a distributed systems engineer who turned to PostgreSQL and (distributed) databases in 2014 by joining Citus Data, after working on Amazon CloudFront, Route 53, and a PhD in cooperation between self-driving cars.

I've been building PostgreSQL extensions since 2015, when Citus unforked and became the first commercially successful extension. My most widely used extension is pg_cron. I also created several proprietary extensions at Microsoft after it acquired Citus Data in 2019, and led Citus development for most of the past 10 years.

I now work for Crunchy Data where we're turning PostgreSQL into a very fast analytics / lakehouse platform via a new set of extensions. Overall, I like creating software products and solving challenging customer problems. I also love doing it through extensions because of all the synergies you automatically create between existing PostgreSQL features and extensions, and because of the simple deployment model.

In my personal time I have a toddler.

**Comment by Florents-Tselai** [comment]
Hi, I'm Florents (Flo), based in Greece.
 
I've been freelancing as a data engineer/architect for 10+ years and always kept pushing for more Postgres (even before it was as cool as it is today). I have been writing extensions for half of that.

When it comes to building Postgres extensions, there are two kinds I usually do:
- The first is **taking a domain-specific algorithm, packaging it as an extension** (usually involves some glue code), and making it available to Postgres users via SQL. I've found this approach very convenient for business users (now their BI analysts can use these algos without waiting for complex pipelines and packages to be installed or for a Python engineer to be available.) This almost always reduces data storage costs and computation costs, too. An example of such a project is https://github.com/Florents-Tselai/vasco 
- The second type of extension I like is **adding support for DSLs in Postgres.** DSLs  The most popular Postgres DSL is jsonpath, but using this model, one can really treat Postgres like a mini-operating system. An example of this is https://github.com/Florents-Tselai/pgJQ which adds jq-syntax support in Postgres. DSLs were popular two decades ago, but one would be surprised to see how popular they still are in (closed) business environments.  There's space for a whole family of possible languages one can embed in Postgres.

For the past few months, I've also been working on making it easier for Python code to be used as a Postgres extension (think plpython3u on steroids). But that's a hobby project for now, so it moves a bit slower. (reach out if you're working on something similar and have ideas).

Last, there's some room & need for streamlining extension boilerplates. When I started, I used to copy-past snippets from larger codebases like Citus and Timescale for pretty standard things. There's also fragmentation there. Extensions are Makefile-based but there's room for templating. For myself, for example, I built this [cookiecutter-template for pg extensions](https://github.com/Florents-Tselai/cookiecutter-postgres-extension) to get myself started more easily.

So that's my mental model around PG extensions. Looking forward to collaborating with you all :).

**Comment by fabriziomello** [comment]
Hi folks, I'm Fabrízio Mello, based in Brazil.

My first contact with Postgres was late 90s in the university and after that never worked in a single IT project without our beloved Slonik. Currently I run one of the oldest Postgres Professional Services company in Brazil named [Timbira](https://timbira.com.br), and also work as a Software Engineer at [Timescale](https://timescale.com) developing the [TimescaleDB extension](https://github.com/timescale/timescaledb).

I've been involved with Brazilian Postgres community for long time now organizing events (PGConf Brazil, PGDays, Meetups) and also managing the community itself. Regarding the global community I already participated of many Postgres events (even as speaker) and already contributed here and there with patches and also reviewing code from others

In my personal time I like to "drive" my grill (also my smoker) to prepare delicious barbecue. Also love to workout and practice Judo.

**Comment by achanda** [comment]
Hi y'all 👋 I'm Abhishek, based in Omaha, Nebraska. I grew up in India, lived in the UK and later moved to the US.

I have worked at a wide array of roles throughout my career. I have been focused on databases and storage over the last couple of years. I work at Cloudflare where we manage a large set of on-prem clusters. I think extensions provide an unique way for people with a diverse set of expertise to contribute to the Postgres ecosystem, hence extensions are very important.

Personally, I love to travel and to play with my twins.

**Comment by keithf4** [comment]
Hello all I'm Keith Fiske based out of Maryland, USA. I'm a Senior Database Engineer at Crunchy Data.

I've been working with PostgreSQL since about version 8.2 and have been an avid extension developer since 9.1 first came out and I started work on https://github.com/pgpartman/pg_partman not too long after. I also maintain the [pgMonitor project ](https://github.com/CrunchyData/pgmonitor) and help to develop automation solutions for PostgreSQL with Ansible.

Outside of work I'm an avid video game player and, after moving to a new house with a lot more space, recently got into gardening and trying to get a food forest going.

**Comment by philippemnoel** [comment]
Hi everyone! 👋

I'm Philippe, based out of New York, USA. I'm one of the founders of [ParadeDB](https://github.com/paradedb/paradedb). We are building full-text search and analytics capabilities in Postgres via extensions. Our goal is to enable organizations to remove/avoid dependency on Elasticsearch, Apache Pinot, etc. We're primarily known for building [pg_search](https://github.com/paradedb/paradedb/tree/dev/pg_search) and [pg_lakehouse](https://github.com/paradedb/paradedb/tree/dev/pg_lakehouse).

I first got introduced to Postgres as a user over the last few years, and took the plunge entering the community and starting ParadeDB in 2023 summer, with my co-founder Ming. Excited to be here, it's going to be a bright future for the extension community.

In my free time, I'm an avid badminton player, boxer and runner.

**Comment by jcflack** [comment]
Hi! I'm Chap, in Lafayette, Indiana, USA for the last not-quite-30 years, probably best known here for maintaining [the PL/Java extension][pljava] since its initial developer, Thomas Hallgren, gave me a commit bit nine years ago. I had been using it for a few years at that point in an internal database at my workplace, and made my first PL/Java bug report in 2013. :)

I have always found databases interesting, though I did not specialize in them at the time of my BA or MSCS. I enjoyed messing around with University INGRES as an undergrad. In the mid-1980s, shortly after graduating and unaware of Postgres, I was poring over the SQL-86 standard at the public library and idly speculating about what would be needed to make the type system extensible. :) A few years later I ended up with my very own commercial INGRES license, during the brief period when SCO bundled it with their Intel desktop Unix. I did not begin working with PostgreSQL until the mid-aughts, after grad school.

As a grad student, I was involved in the construction of a couple research Java virtual machines: directly in [Ovm][], and indirectly in IBM's [Jikes][]. That background, plus my abiding interest in databases, probably explains why I enjoy working on PL/Java. So far, my other contributions to PostgreSQL have been fairly minor, a doc section or tiny patch here or there.

I was in a cycling club for some years, using a bike I had built for touring more than speed, and usually joining the slower rides and enjoying the conversation. Haven't been on a ride for a while. The chamber vocal group I sang in for twenty years had its farewell concert this spring. I've been picking up the absorbing new skills of sprucing up a 1929 bungalow, and dabbling in storytelling. As rewarding activities go, it is hard to beat taking nieces and other young people to theatre and music events.

[pljava]: https://tada.github.io/pljava/
[Ovm]: https://janvitek.org/soft/ovm/
[Jikes]: https://www.jikesrvm.org/

**Comment by daamien** [comment]
Hi I'm Damien Clochard from France

I'm one of the founders of [Dalibo](https://dalibo.com/en/) and main developer of the [PostgreSQL Anonymizer](https://gitlab.com/dalibo/postgresql_anonymizer/) extension.

I'm believe that the PostgreSQL extensions ecosystem will be main driver of success for Postgres in the next ten years. It's also a great way to attract new developers and new ideas in the Postgres community.

Lately I invested some time to rewrite the PostgreSQL Anonymizer in Rust using PGRX and that experience convinced me that Rust is a great language for extensions in terms of stability, performance and development comfort

**Reply by fabriziomello** [reply]
Ohhh... I didn't noticed you rewrote Postgres Anonymizer in Rust... so cool!!! Looking forward for dynamic masking capabilities!!!

**Reply by daamien** [reply]
Thanks ! Version 2.0 is currently in alpha phase, we hope to release it as stable in the forthcoming months

**Comment by theory** [comment]
Greetings, I'm David Wheeler, based in Harlem, New York City. I've been using Postgres since 2000 when we switched from Oracle at my job and eventually opens-sourced Bricolage, a CMS, on Postgres. My first extensions were [pgTAP](https://pgtap.org) in June 2008, and [citext 2](https://www.postgresql.org/docs/current/citext.html), committed in July 2008 — which is wild, because I am not a C programmer!

In 2011 I built [PGXN](https://pgxn.org), the PostgreSQL Extension Network, and continue to run it. You should [release](https://manager.pgxn.org/howto) your extensions there if you haven't already! (Use [pgxn-tools](https://hub.docker.com/r/pgxn/pgxn-tools) in a GitHub workflow to automate releases.) Most of the last 10 years I've spent as a Data Architect and Staff Engineer at a few different companies, but on the first of this year I started work at [Tembo](https://tembo.io), where my time's dedicated to working with the Postgres community to design and build new and improved extension distribution services under the loose umbrella of [PGXN v2](https://wiki.postgresql.org/wiki/PGXN_v2). I'm super stoked about this work, and appreciate learning as much as I can from you all to make the best system possible.

I'm also responsible for [Sqitch](https://sqitch.org), a database change management system, and I write semi-regularly at [Just a Theory](https://justatheory.com).

In my spare time I hack on a side project porting the PostgreSQL SQL/JSON path parser and executor to Go to power an encrypted database, and otherwise enjoy a daily bike ride and traveling abroad with my wife and, on occasion, my college kid.
